File: src/main/java/com/wikia/tibia/enums/ItemClass.java
package com.wikia.tibia.enums;

import com.fasterxml.jackson.annotation.JsonValue;

public enum ItemClass {
    BODY_EQUIPMENT("Body Equipment"),
    WEAPONS("Weapons"),
    RUNES("Runes"),
    HOUSEHOLD_ITEMS("Household Items"),
    PLANTS_ANIMAL_PRODUCTS_FOOD_AND_DRINK("Plants, Animal Products, Food and Drink"),
    TOOLS_AND_OTHER_EQUIPMENT("Tools and other Equipment"),
    OTHER_ITEMS("Other Items");

    private String description;

    ItemClass(String description) {
        this.description = description;
    }

    @JsonValue
    public String getDescription() {
        return description;
    }
}

File: src/main/kotlin/com/wikia/tibia/enums/ItemClass.kt
package com.wikia.tibia.enums

import com.fasterxml.jackson.annotation.JsonValue
import com.wikia.tibia.interfaces.Description

enum class ItemClass(@JsonValue override val description: String) : Description {
    BODY_EQUIPMENT("Body Equipment"),
    WEAPONS("Weapons"),
    RUNES("Runes"),
    HOUSEHOLD_ITEMS("Household Items"),
    PLANTS_ANIMAL_PRODUCTS_FOOD_AND_DRINK("Plants, Animal Products, Food and Drink"),
    TOOLS_AND_OTHER_EQUIPMENT("Tools and other Equipment"),
    OTHER_ITEMS("Other Items");
}
